entgql/extension.go

```diff
@@ -140,12 +173,34 @@ func (e *Extension) Templates() []*gen.Template {
 
 // Hooks of the extension.
 func (e *Extension) Hooks() []gen.Hook {
-	return e.hooks
+	var hooks []gen.Hook
+	if e.cfg != nil || e.doc != nil {
+		hooks = append(hooks, func(next gen.Generator) gen.Generator {
+			return gen.GenerateFunc(func(g *gen.Graph) error {
+				if g.Annotations == nil {
+					g.Annotations = gen.Annotations{}
+				}
+				if e.cfg != nil {
+					g.Annotations[GQLConfigAnnotation] = e.cfg
+				}
+				if e.doc != nil {
+					g.Annotations[GQLSchemaAnnotation] = e.doc
+				}
+				return next.Generate(g)
+			})
+		})
+	}
+	return append(hooks, e.hooks...)
 }
 
-// DefaultMapScalar provides the default mapping from ent.Schema type into GraphQL
-// scalar type. In order to override this function, use the WithMapScalarFunc option.
-func DefaultMapScalar(f *gen.Field, op gen.Op) string {
```

Comment:
@giautm, note that this function was deleted. Please see the documentation for the `WithMapScalarFunc` option.

Reply:
Yeah, It cool. I don't like to import `go-graphql/graphql` in `entc.go` to support default behavior.